controllers/message_util.go

```diff
@@ -224,12 +224,32 @@ func getQuestionWithSuggestions(question string, count int) (string, error) {
 	return question, nil
 }
 
-func parseAnswerAndSuggestions(answer string) (string, []string, error) {
+func parseAnswerAndSuggestions(answer string) (string, []object.Suggestion, error) {
 	parts := strings.Split(answer, divider)
 
+	suggestions := []object.Suggestion{}
+
 	if len(parts) < 2 {
-		return answer, []string{}, nil
+		return answer, suggestions, nil
+	}
+
+	for i := 1; i < len(parts); i++ {
+		suggestions = append(suggestions, object.Suggestion{Text: formatSuggestion(parts[i]), IsHit: false})
```

Comment:
Why not use prompt to let AI return JSON to you? Avoid do parsing and capsulation by yourself

Reply:
mean: {suggestion:xxx,isHit:false} ||| {suggestion:xxx,isHit:false} ?
More complex outputs will increase the probability of AI making mistakes
ant "isHit:false"  may waste unnecessary tokens